### PR TITLE
[WIP] - Test driven development redoing of unpack

### PR DESCRIPTION
**Pull request by VoR0220**
Currently these are basically a ton of tests I've come up with while hacking around and I think I've finally gotten an idea for where to better go from here. There is some reorganization that is needed in the testing framework overall and there needs to be more separation of the code ala how it is done with the packing.go file (and each of these files ought to be individually tested).

### CODE REVIEW

File: accounts/abi/abi.go [hunk 1]
```diff
@@ -34,7 +34,8 @@ import (
 type ABI struct {
 	Constructor Method
 	Methods     map[string]Method
-	Events      map[string]Event
+	// todo: make events loggable from an interface that specifies very abstract functions.
```
Comment (VoR0220) [inline accounts/abi/abi.go]: ignore this

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 2615dc38870a9bce8863e92bbd728f58504f2db7
  * Commit subjects should be kept under 100 characters

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 2615dc38870a9bce8863e92bbd728f58504f2db7
  * Commit subjects should be kept under 100 characters

 * Commit: e4dce0b1312f88b9cc496f2481f946d5dbfd5154
  * Commit subjects should be kept under 100 characters

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 2615dc38870a9bce8863e92bbd728f58504f2db7
  * Commit subjects should be kept under 100 characters

 * Commit: e4dce0b1312f88b9cc496f2481f946d5dbfd5154
  * Commit subjects should be kept under 100 characters

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Pull request closed without merging**